### PR TITLE
Add Safe Repository

### PR DESCRIPTION
**Pull request by fmrsabino**
- Adds `SafeRepository`
  * This repository exposes methods used to interact and retrieve data from a specific Safe
  * Currently it allows routes to get a specific Safe information with `getSafe`

### CODE REVIEW

File: src/domain/safe/entities/schemas/safe.schema.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  properties: {
+    address: { type: 'string' },
+    nonce: { type: 'number' },
+    threshold: { type: 'number' },
+    owners: { type: 'array', items: { type: 'string' } },
+    masterCopy: { type: 'string' },
+    modules: { type: 'array', nullable: true, items: { type: 'string' } },
+    fallbackHandler: { type: 'string' },
+    guard: { type: 'string' },
+    version: { type: 'string', nullable: true },
```
Comment (moisses89) [inline src/domain/safe/entities/schemas/safe.schema.ts:15]: I think that version is not `nullable`.

File: src/domain/safe/safe.repository.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(
+    @Inject(ITransactionApiManager)
+    private readonly transactionApiManager: ITransactionApiManager,
+    private readonly validationErrorFactory: ValidationErrorFactory,
+    private readonly jsonSchemaService: JsonSchemaService,
+  ) {
+    this.isValidSafe = jsonSchemaService.compile(safeSchema);
+  }
+
+  async getSafe(chainId: string, address: string): Promise<Safe> {
```
Comment (moisses89) [inline src/domain/safe/safe.repository.ts:23]: Why chainId is a string?

Comment (fmrsabino) [reply src/domain/safe/safe.repository.ts:23]: This was to be consistent to all the other `chainId` in the project. I can make a separate PR to change all of them

File: src/domain/safe/entities/schemas/safe.schema.ts (hunk 1 above)
Comment (fmrsabino) [reply src/domain/safe/entities/schemas/safe.schema.ts:15]: Yes you are right. The current CGW has it as nullable https://github.com/safe-global/safe-client-gateway/blob/56fa8a32b2284c64af2c5caccb17bfa04aa207fc/src/providers/info.rs#L64

I will change it here.

**Comment by moisses89** [comment]
> * Currently is allows routes to get a specific Safe information with `getSafe`

Nitpick: Currently it allows

File: src/domain/safe/entities/schemas/safe.schema.ts (hunk 1 above)
Comment (fmrsabino) [reply src/domain/safe/entities/schemas/safe.schema.ts]: See [22b5798](https://github.com/5afe/safe-client-gateway-nest/pull/116/commits/22b5798bcd7c6e17e4788c078b3dd1ae0900b639)

**Review by moisses89 — APPROVED** [review]
Looks good for me!

**Pull request merged**